src/lib/utility/html-page.js

```diff
@@ -27,6 +29,40 @@ export default class HTMLPage {
     return (window.self === window.top)
   }
 
+  /**
+   * Checks wither the current browsing content (represented by window object)
+   * is a valid target for a UI controller activation.
+   * The browsing context could be either the topmost window within a browser tab
+   * or a window within a frame that is part of the topmost or any other window.
+   * @returns {boolean} - True if the browsing content is valid, false otherwise.
+   */
+  static get isValidTarget () {
+    // Check if page URL is not excluded
+    for (const url of HTMLPage.targetRequirements.excludedURLs) {
+      if (window.document.URL.search(url) !== -1) {
+        return false
+      }
+    }
+
+    if (!window.document.body) {
+      return false
+    }
+
+    if (window.document.body.clientWidth < HTMLPage.targetRequirements.minWidth) {
```

Comment:
can we just add a TODO comment that this will need to be changed when we add mobile support?

Reply:
Updated